### PR TITLE
imagebuildah: drop resolveSymlink()

### PR DESCRIPTION
**Pull request by nalind**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Drop a bespoke `resolveSymlink()` function and replace it with a call to `copier.Eval()`.

Saving a volume cache using VFS had the side-effect of creating the volume's directory, so have the overlay path for it do so as well.  Have the VFS path add transient mounts to the locations in the rootfs to keep buildah.Run() from doing any saving/restoring of volume contents on its own.

Fix a duplicate test name that newer bats would complain about.

#### How to verify it

Tests should continue to pass.

#### Which issue(s) this PR fixes:

`resolveSymlink()` was interpreting all of its child process's output as the resolved value of the path it was given, which caused #3153.

#### Does this PR introduce a user-facing change?

```
None
```

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM But tests not happy.

**Comment by umohnani8** [comment]
LGTM

**Comment by nalind** [comment]
Had some scope creep here, but tests are passing now.

File: imagebuildah/stage_executor.go [hunk 1]
```diff
@@ -102,16 +102,24 @@ func (s *StageExecutor) Preserve(path string) error {
...
+			return errors.Wrapf(err, "making evaluated path %q relative to %q", evaluated, s.mountPoint)
+		}
+		if strings.HasPrefix(symLink, ".."+string(os.PathSeparator)) {
+			return errors.Errorf("evaluated path %q was not below %q", evaluated, s.mountPoint)
+		}
+		archivedPath = evaluated
+		path = string(os.PathSeparator) + symLink
 	} else {
-		return errors.Wrapf(err, "error reading symbolic link to %q", path)
+		return errors.Wrapf(err, "error evaluating path %q", path)
```
Comment (rhatdan) [inline imagebuildah/stage_executor.go:116]: I worry about a stutter here.  But ...

File: tests/conformance/conformance_test.go [hunk 2]
```diff
@@ -1451,15 +1451,18 @@ var internalTestCases = []testCase{
 		contextDir:      "mount",
 		buildahRegex:    "/tmp/test/file.*?regular file.*?/tmp/test/file2.*?regular file",
 		withoutDocker:   true,
-		transientMounts: []string{"@@TEMPDIR@@:/tmp/test"},
+		transientMounts: []string{"@@TEMPDIR@@:/tmp/test" + selinuxMountFlag()},
```
Comment (rhatdan) [inline tests/conformance/conformance_test.go:1454]: I believe the :Z is already ignored on systems with SELinux disabled, so this code is probably not necessary.

**Pull request merged**